src/figuresClasses.ts

```diff
@@ -1,19 +1,97 @@
-export interface Figure {
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape = Shape.Triangle;
+
+  a: number;
+
+  b: number;
+
+  c: number;
+
+  constructor(
+    public color: Color,
+    a: number,
+    b: number,
+    c: number,
+  ) {
+    if (a >= b + c || a <= Math.abs(b - c) || a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Impossible triangle sides');
+    }
+
+    this.a = a;
+    this.b = b;
+    this.c = c;
+  }
+
+  getArea(): number {
+    const p: number = (this.a + this.b + this.c) / 2;
+    const area = Math.floor(
+      Math.sqrt(p * (p - this.a) * (p - this.b) * (p - this.c))
```

Comment:
I would do destructuring of 'this', it will make your code shorter

Reply:
Good idea!